spec/workers/scheduled_publishing_worker_spec.rb

```diff
@@ -19,6 +22,27 @@
       expect(edition.reload.actions.first.requester.name).to eq robot.name
     end
 
+    it "#perform sends email alerts for an edition" do
+      stub_any_publishing_api_call
+      stub_email_alert_api_accepts_content_change
+
+      ScheduledPublishingWorker.enqueue(edition)
+      travel_to(1.hour.from_now)
+      ScheduledPublishingWorker.drain
+
+      expect(PublishingApiWorker.jobs.size).to eq(1)
+      travel_to(1.hour.from_now)
+      PublishingApiWorker.drain
+      expect(PublishingApiWorker.jobs.size).to eq(0)
+
+      expect(EmailAlertApiWorker.jobs.size).to eq(1)
+      travel_to(1.hour.from_now)
+      EmailAlertApiWorker.drain
+      expect(EmailAlertApiWorker.jobs.size).to eq(0)
```

Comment:
From a run through of the code, I'm not sure the multiple `travel_to`s are necessary for this to pass? I also don't see much value in verifying that the job sizes drop to 0, since `drain` is naturally going to mean that's always true, and indeed we're already not checking that for `ScheduledPublishingWorker`.

I had a think about whether the sequencing / dependencies could be made any clearer - what do you think?

```suggestion
      # Queue up and execute the ScheduledPublishingWorker
      ScheduledPublishingWorker.enqueue(edition)
      travel_to(1.hour.from_now)
      ScheduledPublishingWorker.drain

      # Check that the above created a PublishingApiWorker, then execute it
      expect(PublishingApiWorker.jobs.size).to eq(1)
      PublishingApiWorker.drain

      # Check that the above created a EmailAlertApiWorker, then execute it
      expect(EmailAlertApiWorker.jobs.size).to eq(1)
      EmailAlertApiWorker.drain
```

Reply:
yup i'll rebase it in